charts/agent-control/templates/ac-deployment-secret.yaml

```diff
@@ -0,0 +1,11 @@
+{{- if (index .Values "agent-control-deployment").enabled -}}
```

Comment:
same here, why not doing .Values.agent-control-deployment.enabled? if you are scared of nulls you could do the 'a(.b(.c))' trick

Reply:
As commented on another thread, the problem is with dashes in the name.